### PR TITLE
Automatic generation of documentation

### PR DESCRIPTION
**Pull request by ldez**
- generate a detailed CLI help
```
$ lego dnshelp -c cloudflare
```
- generate a documentation site:
  demo: https://ldez.github.io/lego/
- new readme
  demo: https://github.com/ldez/lego/tree/feature/doc-site

### CODE REVIEW

File: internal/dnsdocs/generator.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
+package main
```
Comment (wyattjoh) [inline internal/dnsdocs/generator.go]: Maybe worth attaching a `//go:generate` comment here to allow `go generate ./...` to trigger this as well?

File: providers/dns/iij/iij.toml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    IIJ_POLLING_INTERVAL = "Time between DNS propagation check"
+    IIJ_PROPAGATION_TIMEOUT = "Maximum waiting time for DNS propagation"
+    IIJ_TTL = "The TTL of the TXT record used for the DNS challenge"
+
+[Links]
+  API = "http://manual.iij.jp/p2/pubapi/http://manual.iij.jp/p2/pubapi/"
+  GoClient = "https://github.com/iij/doapi"
+
+
+
```
[the remark this answers is not in the document]
Comment (ldez) [reply providers/dns/iij/iij.toml]: I didn't find a real linter, I tried several tools but they don't work as expected.
The main problem of these tools is the multiline.

File: internal/dnsdocs/dns.md.tmpl [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+date: 2019-03-03T16:39:46+01:00
+draft: false
+slug: {{ .Code }}
+---
+
+<!-- THIS DOCUMENTATION IS AUTO-GENERATED. PLEASE DO NOT EDIT. -->
+<!-- {{ .GeneratedFrom }} -->
+<!-- THIS DOCUMENTATION IS AUTO-GENERATED. PLEASE DO NOT EDIT. -->
+<!-- {{ .GeneratedFrom }} -->
+<!-- THIS DOCUMENTATION IS AUTO-GENERATED. PLEASE DO NOT EDIT. -->
```
Comment (mholt) [inline internal/dnsdocs/dns.md.tmpl]: Any reason for the repetition?

**Comment by mholt** [comment]
Wow, this is really cool!

**Pull request merged**